Heavily update tcpdump story

ameowlia opened the pull request:
Complete change the tcpdump story. Now on c2c. 

Problems with the old story:
- It needs some work 
- Had more people run through it, and they had problems 
- I don't want to confuse people more with a bad story
- Lost the source code for the pingaling app

Better new story: 
- Now related to c2c (where I use the tool the most in real debugging situations) 
- More help provided to figure out tcpdump 
- Less throwing people in the deep end

cc @christianang

ameowlia committed:
Remove tcpdump story

- It needs some work 
- Had more people run through it, and they had problems 
- I don't want to confuse people more with a bad story
- Will make another commit later with a better tcpdump story

ameowlia committed:
add tcpdump story

- this time for c2c
- more instructions than the last story

bwinter commented:
Wow full rewrite! 

🚧 🚧 🚧 🚧 
🚜 👷‍♀️ 👷 🏗

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+Inspect your c2c packets with tcpdump
+
+### What?
+Sometimes a user says "container to container networking is broken! AppA can't talk to AppB". After making sure that they have c2c policies set, the next thing you might do is use tcpdump to inspect the packets.
```
Might be worth defining `c2c`. Some students (e.g. designers or new Pivots) might not have the context.

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+### What?
+Sometimes a user says "container to container networking is broken! AppA can't talk to AppB". After making sure that they have c2c policies set, the next thing you might do is use tcpdump to inspect the packets.
+
+Tcpdump is a CLI tool that allows you to inspect all of the traffic (not just TCP, despite the name) flowing through your container. This is a great tool to use to debug how far packets are making it, how long the packets are taking to reach their destination, and to look at where in the TCP handshake things are failing.
```
Probably don't need `to use` in the last sentence.

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+
+### How?
+**Setup**
+1. You should have two apps pushed named Boots and Swiper.
```
Maybe "Use the two apps we pushed In the previous story: Boots and Swiper" or something along those lines so that if someone jumps to this story they aren't confused about where these two apps come from?

bwinter commented on networking_monitoring.prolific:
````diff
@@ -97,6 +97,99 @@ L: security
+ ```
+
+**Look at those packets**
+1. In another terminal, ssh onto the Diego Cell where boots is running and become root (see the help section if you don't know how to do this).
````
I like the call out to the help section! 🙏

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+**Look at those packets**
+1. In another terminal, ssh onto the Diego Cell where boots is running and become root (see the help section if you don't know how to do this).
+1. Run `tcpdump`.
+    Ahhhhh too much information! ctrl+c! ctrl+c!  On a Diego Cell there are many packets being sent around, and tcpdump gives information about ALL OF THEM. You need to figure out a way to filter this overwhelming stream of information.
```
I really like how this line makes the story more light hearted.

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+
+**Add c2c policy**
+1. Add c2c policy to allow traffic from Boots to Swiper (`cf add-network-policy --help`)
+1. Continually try to curl Swiper from Boots
```
Might be nice to call out that this was done previously. "You should still be curling Swiper from Boots (if not go back to **Curl swiper from boots** section" (or somethin like that)

bwinter replied:
Or maybe refer to the "in the original terminal you should still be curling swiper form boots."

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+1. Add c2c policy to allow traffic from Boots to Swiper (`cf add-network-policy --help`)
+1. Continually try to curl Swiper from Boots
+
+**Look at those packets**
```
nit: Could also say "inspecting packets" or "inspect packets". Look at those packets is also good, just thinking of other wording that doesn't use "those" (which implies a specific set of packets)

bwinter commented on networking_monitoring.prolific:
````diff
@@ -97,6 +97,99 @@ L: security
+ ```
+cf ssh boots
+ ```
+1. Continually try to curl swiper from boots
````
Noticing inconsistent use of capitol lettering for the app names in sentences. Would be nice to standardize.

bwinter commented on networking_monitoring.prolific:
```diff
@@ -97,6 +97,99 @@ L: security
+[helpful common tcpdump commands](https://www.rationallyparanoid.com/articles/tcpdump.html)
+[debugging non-c2c traffic in CF](https://github.com/cloudfoundry/cf-networking-release/blob/develop/docs/troubleshooting.md#debugging-non-c2c-packets)
+
+L: security
```
This must mean something but I am not sure what 🤔

ameowlia replied:
If you are referencing the "debugging non-c2c traffic"  - that was left over from the tcpdump story around ASGs. I have removed this link. 

If you are referencing the label - all of the other stories in this series have that label so I kept it to be consistent. I'm happy to remove it, if you think that's best.

bwinter replied:
Oh sorry for the ambiguity. I was referring to the `L: security`.

TIL! I didn't know that was part of the DSL in here. Thanks and feel free to keep it! (I definitely don't know everything, just doing what I can to help with PRs in this repo :P)

bwinter reviewed (commented):
Thanks again for putting in the work here!

Definitely looks a lot more "step-by-step" than the previous story. I think that's going to be super useful, especially for non eng students.

ameowlia commented:
Thanks for your thorough review @bwinter. I will update soon. ⭐️  

edited to add: updating done!

ameowlia committed:
update c2c tcpdump story

- reference setup from previous story
- define c2c first
- downcase boots and swiper app names

bwinter reviewed (approved):
Nice job on the quick updates!

I am excited that you were able to get some proxy onboarders to experience the story and give you some feedback. That's such a great way to make these stories valuable!

LGTM!

bwinter commented:
(I'll let you do the honors of merging it if you like!)

ameowlia commented:
@bwinter I don't have permissions. You will need to merge. Thanks!

bwinter commented:
Oh no! I hadn't realized :(. 

Hmmmm, I don't fully understand all the permissioning around this repo. It kind of seems like half the pleasure of making a PR is merging it! (I'll see if I can get you that permission real quick, if it doesn't seem easy I'll merge it just to get your changes on master then deal with alternatives down the road.)

bwinter commented:
I didn't see any obvious ways to allow merge for contributors like your self (I am not a github ninja); so, I just went ahead and added you as a team member who can write. (your helping so much we should just call you what you are :D).

Does it let you merge now?

ameowlia commented:
success! Thanks @bwinter ❄️ !
